ghcide/session-loader/Development/IDE/Session/Diagnostics.hs

```diff
@@ -0,0 +1,86 @@
+module Development.IDE.Session.Diagnostics where
+import           Control.Monad
+import qualified Data.Aeson                        as Aeson
+import           Data.Char                         (isLower)
+import           Data.List
+import           Data.List.Extra                   (dropPrefix, split)
+import           Data.Maybe
+import qualified Data.Text                         as T
+import qualified Data.Vector                       as Vector
+import           Development.IDE.Types.Diagnostics
+import           Development.IDE.Types.Location
+import qualified HIE.Bios.Cradle                   as HieBios
+import           HIE.Bios.Types                    hiding (Log)
+import           System.FilePath
+
+{- | Takes a cradle error, the corresponding cradle and the file path where
+  the cradle error occurred (of the file we attempted to load).
+  Depicts the cradle error in a user-friendly way.
+-}
+renderCradleError :: CradleError -> Cradle a -> NormalizedFilePath -> FileDiagnostic
+renderCradleError (CradleError deps _ec ms) cradle nfp
+  | HieBios.isCabalCradle cradle && any (isInfixOf "Error: cabal: Failed extracting script block:") ms =
+      let (fp, showDiag, diag) = ideErrorWithSource (Just "cradle") (Just DiagnosticSeverity_Error) nfp $ T.unlines $ map T.pack userFriendlyMessage in
+        (fp, showDiag, diag{_data_ = Just (Aeson.Array $ Vector.fromList $ map (Aeson.String . T.pack) absDeps)})
```

Comment:
I do think it's worth defining a datatype that you use here with derived to/fromJSON instances, even if it's ultimately just a list of strings. Just makes it harder to mess up.